src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -61,8 +62,9 @@
         this.number = pr.getNumber();
         this.target = new BranchSCMHead(pr.getBase().getRef());
         // the source stuff is immutable for a pull request on github, so safe to store here
-        this.sourceOwner = pr.getHead().getRepository().getOwnerName();
-        this.sourceRepo = pr.getHead().getRepository().getName();
+        GHRepository repository = pr.getHead().getRepository(); // may be null for deleted forks JENKINS-41246
+        this.sourceOwner = repository.getOwnerName();
```

Comment:
Trying to ascertain reasons about code I wrote more than 15 minutes ago is an exercise in futility... I suspect I did mean to check for `null` though... good catch